### PR TITLE
Add Rust queries

### LINKED ISSUE #52: No locals.scm query found for c, cpp and rust
```
health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `git` executable found.
  - OK: `cc` executable found.

## c parser healthcheck
  - OK: c parser found.
  - WARNING: No `locals.scm` query found for c
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `highlights.scm` query found for c
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## cpp parser healthcheck
  - OK: cpp parser found.
  - WARNING: No `locals.scm` query found for cpp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `highlights.scm` query found for cpp
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## lua parser healthcheck
  - OK: lua parser found.
  - OK: `locals.scm` found.
  - OK: `highlights.scm` found.

## rust parser healthcheck
  - OK: rust parser found.
  - WARNING: No `locals.scm` query found for rust
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter
  - WARNING: No `highlights.scm` query found for rust
    - ADVICE:
      - Open an issue at https://github.com/nvim-treesitter/nvim-treesitter

## Missing parsers
  - WARNING: Some parsers are not installed:
    csharp
    html
    typescript
    swift
    java
    python
    nix
    yaml
    elm
    vue
    ruby
    ocaml
    go
    scala
    haskell
    json
    toml
    javascript
    css
    julia
    markdown
    php
    bash
    tsx
    - ADVICE:
      - Install them using `:TSInstall language

```

```
NVIM v0.5.0-504-g55b62a937
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /Library/Developer/CommandLineTools/usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/Users/yashladha/nvim/build/config -I/Users/yashladha/nvim/src -I/Users/yashladha/nvim/.deps/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -I/usr/local/opt/gettext/include -I/Users/yashladha/nvim/build/src/nvim/auto -I/Users/yashladha/nvim/build/include
Compiled by yashladha@yashLadha.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```

### PR DESCRIPTION
**Pull request by vigoux**
Also slightly changes the way constructors are highlighted to better match their highlighting by default.

This will partially fix #52. 

TODO:
- [x] `highlights.scm`
- [ ] `locals.scm`
- [ ] Update `README`

### CODE REVIEW

**Comment by chmln** [comment]
Would love to see this. Please let me know if any help is needed :)

**Comment by kyazdani42** [comment]
Help is needed indeed, the highlights.scm could have some more improvements but is mostly ok, but support for the locals query would be great !

**Comment by chmln** [comment]
I took a shot at the queries, with help from existing implementations for other languages and the playground. I thought it'd be good to submit it upstream as well. Please copy the file from https://github.com/tree-sitter/tree-sitter-rust/pull/85/

P.S. what tools could be used to verify these queries? Perhaps I missed something in the tree-sitter CLI.

**Comment by vigoux** [comment]
Thanks for that! I will look into it soon and give you some feedback!

**Comment by vigoux** [comment]
@chmln you might want to hang out here if you have some questions [Gitter](https://gitter.im/nvim-treesitter/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)

File: queries/rust/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+(type_arguments
+  "<" @punctuation.bracket
+  ">" @punctuation.bracket)
+(type_parameters
+  "<" @punctuation.bracket
+  ">" @punctuation.bracket)
+
+"::" @punctuation.delimiter
+"." @punctuation.delimiter
+";" @punctuation.delimiter
```
Comment (theHamsta) [inline queries/rust/highlights.scm:76]: ", " is missing

**Comment by vigoux** [comment]
I guess that we can add the `highlights.scm` file only for now, I will add the other queries later, when I have more time.

**Comment by vigoux** [comment]
I tried GH conflict resolution thing, and that created a merge commit, I guess it'll need some more testing then....

File: queries/rust/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+(float_literal) @float
+
+(escape_sequence) @string.escape
+
+[
+"as"
+"*"
+"&"
+"'"
+ ] @operator
```
Comment (theHamsta) [inline queries/rust/highlights.scm:165]: a lot of operators are missing:
math stuff, "=>", "?"

can be added in followup PR

File: queries/rust/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+"default"
+"dyn"
+"enum"
+"extern"
+"fn"
+"impl"
+"let"
+"macro_rules!"
+"match"
+"mod"
```
Comment (theHamsta) [inline queries/rust/highlights.scm:111]: @import ? like all other module related things

**Comment by vigoux** [comment]
Yes we can go with that, still some issues with enum constructors and enum destructuring, but that will do for some time. I am hoping to get that working soon.

**Pull request merged**